lib/src/path_parser.cc

```diff
@@ -143,14 +143,18 @@ namespace hocon {
             return token_list { t };
         }
 
+        // The split iterator tacks an empty string onto the end of the iterator if the string
+        // ends in the split character; so we need to remove trailing '.'
+        string text_to_split = boost::trim_right_copy_if(token_text, is_any_of("."));
```

Comment:
And if we don't need that code, we could just drop the const on token_text and trim it in-place.